Modules/Bridge/NumPy/wrapping/PyBuffer.i.in

```diff
@@ -74,18 +74,18 @@
         *np.reshape*.
         """
 
-        assert ndarr.ndim in ( 1, 2, 3, 4 ), \
-            "Only arrays of 1, 2, 3 or 4 dimensions are supported."
+        assert ndarr.ndim in (1, 2, 3, 4, 5), \
+            "Only arrays of 1, 2, 3, 4 or 5 dimensions are supported."
         if not ndarr.flags['C_CONTIGUOUS'] and not ndarr.flags['F_CONTIGUOUS']:
             ndarr = np.ascontiguousarray(ndarr)
 
-        if ( ndarr.ndim == 3 and is_vector ) or (ndarr.ndim == 4):
-            if( ndarr.flags['C_CONTIGUOUS'] ):
-                imgview = itkPyBuffer@PyBufferTypes@._GetImageViewFromArray( ndarr, ndarr.shape[-2::-1], ndarr.shape[-1] )
+        if is_vector and (ndarr.ndim == 3 or ndarr.ndim == 4):
```

Comment:
We are changing the behavior of the case that has `not is_vector and ndar.ndom == 4`, which used to be treated as if `is_vector` were true even when that flag was false.  Will that break anyone?

Perhaps more importantly, we are setting ourselves up for the same issue for `not is_vector and ndar.ndom == 5`, which will be treated as if `is_vector` is true in this new implementation whether or not it is.  If we later accept more than 5 dimensions, will we be breaking the uses that trust us on this new implementation?  I am thinking that we should require that `is_vector` be true if `ndar.ndim == 5` so that users won't be sloppy and leave `is_vector` as false when `ndar.ndim == 5`.

Reply:
Sorry, I mean that the `ndar.ndim == 5` case is ignoring the `is_vector` flag, by always interpreting it as `false` ... and that could be an issue in the future.

Reply:
Yes, we did not have 4D binary packages before this release, but if users built their own packages and did not set `is_vector`, they could experience different behavior. However, this is required for 4D vector support.  Yes, we should avoid this situation if possible for additional dimensional and require users to set `is_vector` explicitly when that is desired. I will remove the `ndim` checks in the condition.